streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -52,31 +61,78 @@
     private final Logger log;
     private final String logPrefix;
 
+    private final StreamThread.ProcessingMode processingMode;
     private final Producer<byte[], byte[]> producer;
-    private final boolean eosEnabled;
 
     private boolean transactionInFlight = false;
     private boolean transactionInitialized = false;
 
-    public StreamsProducer(final Producer<byte[], byte[]> producer,
-                           final boolean eosEnabled,
+    public StreamsProducer(final StreamsConfig config,
+                           final String threadId,
+                           final KafkaClientSupplier clientSupplier,
+                           final StreamThread.ProcessingMode processingMode,
```

Comment:
We can get the processing mode from the first parameter `config` right?

Reply:
+1